src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -3071,6 +3071,18 @@ tx_gateway_frontend::get_ongoing_tx(
                 co_return tx_errc::invalid_txn_state;
             }
 
+            if (old_tx.status == tm_transaction::tx_status::ongoing) {
+                if (old_tx.tx_seq() == tx.tx_seq()) {
+                    co_return tx;
+                } else if (old_tx.tx_seq() != tx.tx_seq() + 1) {
+                    vlog(
+                      txlog.warn,
+                      "fetched a tx with unexpected tx_seq:{}",
+                      old_tx.tx_seq);
+                    co_return tx_errc::unknown_server_error;
```

Comment:
please log both tx_seqs and the context similar to `vlog(txlog.warn, "fetched tx:{} pid:{} etag:{} should have pid:{}", tx.id, old_tx.pid, old_tx.etag, tx.pid);`

also return `invalid_txn_state`